internal/proxy/proxy_test.go

```diff
@@ -241,6 +259,50 @@ func TestClientInitialization(t *testing.T) {
 	}
 }
 
+func TestClientLimitsMaxConnections(t *testing.T) {
+	in := &proxy.Config{
+		Addr: "127.0.0.1",
+		Port: 5000,
+		Instances: []proxy.InstanceConnConfig{
+			{Name: "proj:region:pg"},
+		},
+		MaxConnections: 1,
+	}
+	d := &fakeDialer{}
+	c, err := proxy.NewClient(context.Background(), d, &cobra.Command{}, in)
+	if err != nil {
+		t.Fatalf("proxy.NewClient error: %v", err)
+	}
+	defer c.Close()
+	go c.Serve(context.Background())
+
+	conn1, err1 := net.Dial("tcp", "127.0.0.1:5000")
+	if err1 != nil {
+		t.Fatalf("net.Dial error: %v", err1)
+	}
+	defer conn1.Close()
+
+	conn2, err2 := net.Dial("tcp", "127.0.0.1:5000")
+	if err2 != nil {
```

Comment:
When the server is too busy, this is the call that should fail. not `conn.Read()`

Reply:
The discussion above addresses this.